### PR TITLE
SONAR-6233 Fix behavior of facet on uncharacterized rules

### PR DESCRIPTION
**Pull request by jblievremont**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java [hunk 1]
```diff
@@ -314,6 +317,31 @@ protected void writeFacets(Request request, QueryContext context, Result<?> resu
...
+      if (characValues == null) {
+        return;
+      }
+
+      long mergedCount = 0L;
+      Iterator<FacetValue> characIterator = characValues.iterator();
+      while (characIterator.hasNext()) {
+        FacetValue characValue = characIterator.next();
+        if ("".equals(characValue.getKey()) || DebtCharacteristic.NONE.equals(characValue.getKey())) {
+          mergedCount += characValue.getValue();
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: Far better !

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java [hunk 2]
```diff
@@ -314,6 +317,31 @@ protected void writeFacets(Request request, QueryContext context, Result<?> resu
...
+      Collection<FacetValue> characValues = results.getFacetValues(RuleIndex.FACET_DEBT_CHARACTERISTICS);
+      if (characValues == null) {
+        return;
+      }
+
+      long mergedCount = 0L;
+      Iterator<FacetValue> characIterator = characValues.iterator();
+      while (characIterator.hasNext()) {
+        FacetValue characValue = characIterator.next();
+        if ("".equals(characValue.getKey()) || DebtCharacteristic.NONE.equals(characValue.getKey())) {
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: Ok, you don't like Strings.isEmptyOrNul()... :(

**Comment by julienlancelot** [comment]
LGTM

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java (hunk 2 above)
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: No, it's just that ``""`` is the actual expected value for the ``missing`` aggregation :)

**Pull request merged**